sdk/src/main/scala/com/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -77,6 +78,41 @@ class AccountForgeMessageBuilder(mainchainSynchronizer: MainchainSynchronizer,
     }
   }
 
+  def calculateBaseFee(nodeView: View, parentId: BlockId): BigInteger = {
+    // If the current block is the first block, return the InitialBaseFee.
+    if (nodeView.history.getCurrentHeight < 2) {
```

Comment:
we may use `parentId != params.sidechainGenesisBlockParentId` to check if the new block is genesis block.
we can't rely on the `nodeView.history.getCurrentHeight` results here.
But in general, it is even not a case, since during the verification phase in the validator you may check if `history.isGenesisBlock(...)` before running `calculateBaseFee`

Reply:
I chose the first suggestion, because as you wrote above, we can use this function in several places